### PR TITLE
New logging view for SplitPDFLog

### PR DESCRIPTION
**Pull request by jenachoi**
screencapture-0-0-0-0-3000-en-assignments-7-exam_templates-view_logs-1498146790024

It should work along with #3074. Feedback would be appreciated

### CODE REVIEW

**Comment by david-yz-liu** [comment]
The big comment is that this doesn't show enough information. All fields should be shown for each log object.

File: app/views/exam_templates/_split_pdf_log.html.erb [hunk 1]
```diff
@@ -0,0 +1,28 @@
+<div>
+  <% current_incomplete_num_groups = split_pdf_log.num_groups_in_incomplete %>
+  <% if current_incomplete_num_groups == previous_incomplete_num_groups && split_pdf_log.num_pages_qr_scan_error == 0 %>
+    <p class="log-green-message">
+      <%= "#{t('exam_templates.split_pdf_log.log')} ##{i}" %> - <%= split_pdf_log.uploaded_when.strftime("%d %b %y at %I:%M:%S %p") %>
```
Comment (david-yz-liu) [inline app/views/exam_templates/_split_pdf_log.html.erb]: Reuse an existing date format.

File: app/views/exam_templates/_split_pdf_log.html.erb [hunk 2]
```diff
@@ -0,0 +1,28 @@
...
+    </p>
+    <p><%= t('exam_templates.split_pdf_log.success_log', filename: split_pdf_log.filename, original_num_pages: split_pdf_log.original_num_pages) %>.</p>
+  <% else %>
+    <p class="log-red-message">
+      <%= "#{t('exam_templates.split_pdf_log.log')} ##{i}" %> - <%= split_pdf_log.uploaded_when.strftime("%d %b %y at %I:%M:%S %p") %>
+    </p>
+    <% if split_pdf_log.num_pages_qr_scan_error > 0 %>
+        <p><%= t('exam_templates.split_pdf_log.failure_general_log', filename: split_pdf_log.filename, original_num_pages: split_pdf_log.original_num_pages, num_pages_qr_scan_error: split_pdf_log.num_pages_qr_scan_error) %>.</p>
```
Comment (david-yz-liu) [inline app/views/exam_templates/_split_pdf_log.html.erb]: Use 2 spaces for indentation.

File: app/views/exam_templates/_split_pdf_log.html.erb [hunk 3]
```diff
@@ -0,0 +1,28 @@
...
+    <% else %>
```
Comment (david-yz-liu) [inline app/views/exam_templates/_split_pdf_log.html.erb]: You should have two independent `if` blocks here. Both error messages can appear.

File: app/views/exam_templates/view_logs.html.erb [hunk 4]
```diff
@@ -0,0 +1,17 @@
+<% content_for :title, "#{@assignment.short_identifier} #{SplitPdfLog.model_name.human.pluralize}" %>
+
+<% @heading_buttons = [
+    { link_text: t('exam_templates.back_to_exam_template_page'),
+      link_path: assignment_exam_templates_path }
+] %>
+
+<% i = 1 %>
+<% previous_incomplete_num_groups = @split_pdf_logs[0].num_groups_in_incomplete %>
+<% @split_pdf_logs.each do |split_pdf_log| %>
```
Comment (david-yz-liu) [inline app/views/exam_templates/view_logs.html.erb]: There's an `Enumerable` method that will allow you to loop through the collection and have variables for both the element (exam template) and an index (i).

File: app/views/exam_templates/view_logs.html.erb [hunk 5]
```diff
@@ -0,0 +1,17 @@
+<% content_for :title, "#{@assignment.short_identifier} #{SplitPdfLog.model_name.human.pluralize}" %>
+
+<% @heading_buttons = [
+    { link_text: t('exam_templates.back_to_exam_template_page'),
+      link_path: assignment_exam_templates_path }
+] %>
+
+<% i = 1 %>
+<% previous_incomplete_num_groups = @split_pdf_logs[0].num_groups_in_incomplete %>
```
Comment (david-yz-liu) [inline app/views/exam_templates/view_logs.html.erb]: This doesn't seem like the right comparison to make. For now why don't you just detect whether there are any incomplete groups at all.

**Comment by david-yz-liu** [comment]
Also please do a pull from master.

File: app/views/exam_templates/_split_pdf_log.html.erb (hunk 1 above)
Comment (jenachoi) [inline app/views/exam_templates/_split_pdf_log.html.erb]: `split_pdf_log.uploaded_when` shows a date format like this `2017-06-23 16:56:58 -0400`. I thought 0400 at the end was unnecessary, so this time I changed it slightly to make it look like this `2017-06-23 09:56:58 AM`. But please let me know if I should keep it as it is.

**Pull request merged**